builder.go

```diff
@@ -128,7 +128,13 @@ func (s *VolumeSet) Add(path string) bool {
 	path = strings.TrimSuffix(path, "/")
 	var adjusted []string
 	for _, p := range *s {
-		if p == path || strings.HasPrefix(path, p+"/") {
+		if p == path {
+			// if path is already in the set
+			// pretend its added and return
```

Comment:
nit of a nit of a nit.  Only worry about if you have other changes.
```suggestion
			// pretend it's added and return
```